### PR TITLE
HPCC-24359 Submodule spark plugin third party code

### PR DESCRIPTION
**Pull request by Michael-Gardner**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by Michael-Gardner** [comment]
@xwang2713 please review

**Comment by richardkchapman** [comment]
Is there a good reason to target 7.10.x rather than master here?

File: plugins/spark/CMakeLists.txt [hunk 1]
```diff
@@ -33,6 +33,23 @@ if(SPARK)
 
     option(DOWNLOAD_SPARK_JARS OFF "Download most recent spark plugin java packages at build time")
 
+    #check submodules
+    message(STATUS "Checking for submodule files:")
+    message(STATUS "${CMAKE_CURRENT_SOURCE_DIR}/spark-hadoop/LICENSE")
+    if(NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/spark-hadoop/LICENSE")
+	message(STATUS "Spark-hadoop submodule not populated on disk")
+	message(STATUS "Run: git submodule update --init --recursive")
+        message(FATAL_ERROR "spark-hadoop submodules error")
```
Comment (xwang2713) [inline plugins/spark/CMakeLists.txt:42]: indentation

Comment (xwang2713) [reply plugins/spark/CMakeLists.txt:42]: Other looks good

**Comment by xwang2713** [comment]
Looks good

File: plugins/spark/CMakeLists.txt (hunk 1 above)
Comment (Michael-Gardner) [reply plugins/spark/CMakeLists.txt]: Resolved that indentation issue. Thanks

**Comment by Michael-Gardner** [comment]
@richardkchapman ready for merge into master.

**Pull request merged**